Alignment/OfflineValidation/bin/DMRtrends.cc

```diff
@@ -1,1432 +1,249 @@
- */
-
-vector<int> runlistfromlumifile(TString lumifile) {
-  vector<pair<int, double>> lumiperRun;
-
-  std::ifstream infile(lumifile.Data());
-  int run;
-  double runLumi;
-  while (infile >> run >> runLumi) {
-    lumiperRun.push_back(make_pair(run, runLumi));
-  }
-  size_t nRuns = lumiperRun.size();
-  vector<int> xRunFromLumiFile;
-  for (size_t iRun = 0; iRun < nRuns; iRun++)
-    xRunFromLumiFile.push_back(lumiperRun[iRun].first);
-  return xRunFromLumiFile;
-}
-
-/*! \fn checkrunlist
- *  \brief Check whether all runs of interest are present in the luminosity per run txt file and whether all IOVs analized have been correctly processed
- */
+    Variables.push_back("median");
 
-bool checkrunlist(vector<int> runs, vector<int> IOVlist) {
-  vector<int> missingruns;  //runs for which the luminosity is not found
-  vector<int> lostruns;     //IOVs for which the DMR were not found
-  bool problemfound = false;
-  std::sort(runs.begin(), runs.end());
-  std::sort(IOVlist.begin(), IOVlist.end());
-  if (!IOVlist.empty())
-    for (auto IOV : IOVlist) {
-      if (find(runs.begin(), runs.end(), IOV) == runs.end()) {
-        problemfound = true;
-        lostruns.push_back(IOV);
-      }
-    }
-  if (problemfound) {
-    if (!lostruns.empty()) {
-      cout << "WARNING: some IOVs where not found among the list of available DMRs" << endl
-           << "List of missing IOVs:" << endl;
-      for (int lostrun : lostruns)
-        cout << to_string(lostrun) << " ";
-      cout << endl;
-    }
-  }
-  return problemfound;
-}
-
-/*! \fn DMRtrends
- *  \brief Create and plot the DMR trends.
- */
-
-void DMRtrends(vector<int> IOVlist,
-               vector<string> Variables,
-               vector<string> labels,
-               TString Year,
-               string myValidation,
-               vector<string> geometries,
-               vector<Color_t> colours,
-               TString outputdir,
-               bool pixelupdate,
-               vector<int> pixelupdateruns,
-               bool showlumi,
-               TString lumifile,
-               bool FORCE) {
-  fs::path path(outputdir.Data());
-  if (!(fs::exists(path))) {
-    cout << "WARNING: Output directory (" << outputdir.Data() << ") not found, it will be created automatically!"
-         << endl;
-    //	exit(EXIT_FAILURE);
-    fs::create_directory(path);
-    if (!(fs::exists(path))) {
-      cout << "ERROR: Output directory (" << outputdir.Data() << ") has not been created!" << endl
-           << "At least the parent directory needs to exist, please check!" << endl;
-      exit(EXIT_FAILURE);
-    }
-  }
   TString LumiFile = getenv("CMSSW_BASE");
-  if (lumifile.BeginsWith("/"))
-    LumiFile = lumifile;
+  if (lumiInputFile.BeginsWith("/"))
+    LumiFile = lumiInputFile;
   else {
     LumiFile += "/src/Alignment/OfflineValidation/data/";
-    LumiFile += lumifile;
+    LumiFile += lumiInputFile;
   }
   fs::path pathToLumiFile = LumiFile.Data();
```

Comment:
As we have discussed I will leave it as it is on the whole. I have made some minor changes though. Now lumiInputFile is specified in the style section and not the validation section. If you want to for whatever reason use a lumi file that's not in the /src/Alignment/OfflineValidation/data/ directory you can give the full path starting with /. If you want to just use one of the lumi files in the data directory you only have to give the name of the file. If you don't specify anything it will pick up lumiPerRun_Run2.txt. A check is done to verify that the lumi file contains the appropriate year. The year is set to Run2 by default. A check is also done to verify the file exists. The luminosity type is now read from the file as well. If the file contains the word "delivered" the lumi axis in the trend plots will show delivered. Otherwise recorded is set by default.